packages/data/tests/unit/services/navi-metadata-test.ts

```diff
@@ -0,0 +1,266 @@
+
+    assert.equal(this.service.loadedDataSources.size, 0, '`elideOne` data source is initially not loaded');
+    await this.service.loadMetadata({ dataSourceName: 'elideOne' });
+
+    const tables = keg.all('metadata/table', 'elideOne');
+    assert.ok(
+      tables.every(table => table instanceof TableMetadataModel),
+      '`elideOne` `TableMetadataModel`s are loaded into the keg'
+    );
+    assert.deepEqual(tables.mapBy('id'), ['table0', 'table1'], 'All `elideOne` tables are loaded');
+
+    const dimensions = keg.all('metadata/dimension', 'elideOne');
+    assert.ok(
+      dimensions.every(dim => dim instanceof DimensionMetadataModel),
+      '`elideOne` `DimensionMetadataModel`s are loaded into the keg'
+    );
+    assert.deepEqual(
+      dimensions.mapBy('id'),
+      ['dimension0', 'dimension1', 'dimension2', 'dimension3', 'dimension4'],
+      'All `elideOne` dimensions are loaded'
+    );
+
+    const timeDimensions = keg.all('metadata/timeDimension', 'elideOne');
+    assert.ok(
+      timeDimensions.every(dim => dim instanceof TimeDimensionMetadataModel),
+      '`elideOne` `TimeDimensionMetadataModel`s are loaded into the keg'
+    );
+    assert.deepEqual(
+      timeDimensions.mapBy('id'),
+      ['timeDimension0', 'timeDimension1'],
+      'All `elideOne` time dimensions are loaded'
+    );
+
+    const metrics = keg.all('metadata/metric', 'elideOne');
+    assert.ok(
+      metrics.every(metric => metric instanceof MetricMetadataModel),
+      '`elideOne` `MetricMetadataModel`s are loaded into the keg'
+    );
+    assert.deepEqual(metrics.mapBy('id'), ['metric0', 'metric1'], 'All `elideOne` metrics are loaded');
+
+    const metricFunctions = keg.all('metadata/metric-function', 'elideOne');
+    assert.ok(
+      metricFunctions.every(fn => fn instanceof MetricFunctionMetadataModel),
+      '`elideOne` `MetricFunctionsMetadataModel`s are loaded into the keg'
+    );
+    assert.deepEqual(metricFunctions.mapBy('id'), [], 'All `elideOne` metric functions are loaded');
+
+    assert.ok(this.service.loadedDataSources.has('elideOne'), '`elideOne` data source is loaded');
+  });
+
+  test('loadMetadata - default data source', async function(this: Context, assert) {
+    assert.notOk(this.service.loadedDataSources.has('bardOne'), '`bardOne` data source is initially not loaded');
+    await this.service.loadMetadata();
+    assert.ok(this.service.loadedDataSources.has('bardOne'), 'default data source is loaded');
+  });
+
+  test('loadMetadata - multiple calls', async function(this: Context, assert) {
+    assert.expect(1);
+
+    this.server.urlPrefix = `${config.navi.dataSources[0].uri}/v1`;
+    this.server.get('/tables', function() {
+      assert.ok(true, 'initial metadata load executes a request');
+      return { tables: [] };
+    });
+    await this.service.loadMetadata();
+
+    this.server.get('/tables', function() {
+      assert.notOk(true, 'after metadata is loaded, a fetch request is not executed');
+    });
```

Comment:
```suggestion
    });
    await this.service.loadMetadata();
```

make another call to show it's not fetching?

Reply:
Oops, yeah that was the intention